### PR TITLE
board: add support for the nucleo_f303k8 board

### PR DESCRIPTION
**Pull request by Gimpel-K67**
This branch adds support for the nucleo_f303k8 board, the soc stmf303x8 defconfig and the stm32f303X8 dtsi file.
The configuration is based on the nucleo_l011k4, the f302r8 and the ST reverence manual.

I had successfully tested the following sample code:
blinky
blink_led (uses TIM2_CH1 on PA0)
button
hello_world

Signed-off-by: Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>

### CODE REVIEW

File: boards/arm/nucleo_f303k8/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# NUCLEO-32 F303k8 board configuration
+
+# Copyright (c) 2020 Sebastian Schwabe, Fabian Paschke <sebastian.schwabe@mailbox.tu-dresden.de, fabian.paschke@eas.iis.fraunhofer.de>
```
Comment (erwango) [inline boards/arm/nucleo_f303k8/Kconfig.board:3]: Nit: Long line

File: boards/arm/nucleo_f303k8/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# NUCLEO-32 F303k8 board configuration
+
+# Copyright (c) 2020 Sebastian Schwabe, Fabian Paschke <sebastian.schwabe@mailbox.tu-dresden.de, fabian.paschke@eas.iis.fraunhofer.de>
```
Comment (erwango) [inline boards/arm/nucleo_f303k8/Kconfig.defconfig:3]: Ditto

File: boards/arm/nucleo_f303k8/nucleo_f303k8.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  - xtools
+ram: 12
+flash: 64
+supported:
+  - i2c
+  - spi
+  - gpio
+  - pwm
+  - counter
+  - adc
```
Comment (erwango) [inline boards/arm/nucleo_f303k8/nucleo_f303k8.yaml:17]: i2c, spi, adc not configured in dts

File: dts/arm/st/f3/stm32f303X8.dtsi [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#include <st/f3/stm32f303.dtsi>
+
+/ {
+	ccm0: memory@10000000 {
+		compatible = "st,stm32-ccm";
+		reg = <0x10000000 DT_SIZE_K(8)>;
+	};
+
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(40)>;
```
Comment (erwango) [inline dts/arm/st/f3/stm32f303X8.dtsi:17]:
```suggestion
		reg = <0x20000000 DT_SIZE_K(12)>;
```

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this addition!
I only have few comments.
Also, please split your PR into boards and dts/soc commits.

**Comment by TeckKitty** [comment]
Thanks for the additions. I split it up to three commits for soc, dts and board. It took me some time to fix the format errors but now everything look fine. I also fixed the other things you pointed out.

File: boards/arm/nucleo_f303k8/Kconfig.board [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# NUCLEO-32 F303k8 board configuration
+
+# Copyright (c) 2020 Sebastian Schwabe
+# <sebastian.schwabe@mailbox.tu-dresden.de>
+# Fabian Paschke
+# <fabian.paschke@eas.iis.fraunhofer.de>
```
Comment (erwango) [inline boards/arm/nucleo_f303k8/Kconfig.board:6]: Something like follows would be nicer, but ok.
```
# Copyright (c) 2020 Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>
                                  Fabian Paschke <fabian.paschke@eas.iis.fraunhofer.de>
```

File: soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8 [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# ST Microelectronics STM32F303x8 MCU
+
+#  Copyright (c) 2020 Sebastian Schwabe 
+#							<sebastian.schwabe@mailbox.tu-dresden.de>,
+# 					  Fabian Paschke
+#					   		<fabian.paschke@eas.iis.fraunhofer.de>
```
Comment (erwango) [inline soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8]:
```suggestion
# Copyright (c) 2020 Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>
#                    Fabian Paschke <fabian.paschke@eas.iis.fraunhofer.de>
```

File: soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8 [hunk 7]
```diff
@@ -1,9 +1,9 @@
 # ST Microelectronics STM32F303x8 MCU
 
-#  Copyright (c) 2020 Sebastian Schwabe 
-#							<sebastian.schwabe@mailbox.tu-dresden.de>,
-# 					  Fabian Paschke
-#					   		<fabian.paschke@eas.iis.fraunhofer.de>
+#  Copyright (c) 2020 Sebastian Schwabe
+#  <sebastian.schwabe@mailbox.tu-dresden.de>
+#  Fabian Paschke
+#  <fabian.paschke@eas.iis.fraunhofer.de>
```
Comment (erwango) [inline soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8]: Please fix this in the initial commit

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
@TeckKitty Building ADC test (tests/drivers/adc/adc_api/) requires a modification to the test.
This is not done and this is what is making CI (buildkite: https://buildkite.com/zephyr/zephyr/builds/15942) complaining.
Either you need to update the test, either you need to remove adc from the supported list in the yaml file (but you don't need to remove adc in the dts file).

File: soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8 (hunk 7 above)
Comment (TeckKitty) [reply soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8]: How can I change an outdated commit? I removed the initial commit and add the three new one for dts, soc and board.

**Comment by TeckKitty** [comment]
Ok I try do to the fixes you suggested to me. I mixed up the dts and soc files in one commit. Now it should be correctly separated. 
I removed the ADC from the yaml file because I have no clue how to change the test. 
All checks now passed.
It is my first time doing a merge request so thanks for your patience.

**Review by FRASTM — APPROVED** [review]

File: soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8 (hunk 7 above)
Comment (erwango) [reply soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8]: https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html

TL;DR:
-add a new commit ths fixes the issue. As commit title, use: `fixup! <sha of the commit to be fixed>`
-Repeat for each fix to be done on the branch
-Once you're done: `git rebase -i --autosquash HEAD~<numbers of commits of your branch>`

**Review by erwango — APPROVED** [review]
